test/view/sliding.cpp

```diff
@@ -0,0 +1,152 @@
+// Range v3 library
+//
+//  Copyright Eric Niebler 2014
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+
+#include <list>
+#include <vector>
+#include <forward_list>
+#include <range/v3/core.hpp>
+#include <range/v3/view/iota.hpp>
+#include <range/v3/view/sliding.hpp>
+#include <range/v3/view/reverse.hpp>
+#include <range/v3/view/repeat.hpp>
+#include <range/v3/view/cycle.hpp>
+#include "../simple_test.hpp"
+#include "../test_utils.hpp"
+
+int main()
+{
+    using namespace ranges;
+
+    {
+        std::vector<int> v = view::iota(0,7);
+        auto rng = v | view::sliding(3);
+        ::models<concepts::RandomAccessRange>(rng);
+        ::models<concepts::SizedRange>(rng);
+        auto it = ranges::begin(rng);
+        ::check_equal(*it++, {0,1,2});
+        ::check_equal(*it++, {1,2,3});
+        ::check_equal(*it++, {2,3,4});
+        ::check_equal(*it++, {3,4,5});
+        ::check_equal(*it++, {4,5,6});
+        ::check_equal(*it++, {5,6});
+        ::check_equal(*it++, {6});
```

Comment:
For `view::sliding(N)`, there will be N-1 elements at the end that have fewer than N elements. But not at the start. Why are these the correct semantics?